install/kubernetes/cilium/values.yaml.tmpl

```diff
@@ -1699,13 +1704,204 @@ dashboards:
+  extraEnv: []
+
+  # -- Additional envoy hostPath mounts.
+  extraHostPathMounts: []
+    # - name: host-mnt-data
+    #   mountPath: /host/mnt/data
+    #   hostPath: /mnt/data
+    #   hostPathType: Directory
+    #   readOnly: true
+    #   mountPropagation: HostToContainer
+
+  # -- Additional envoy volumes.
+  extraVolumes: []
+
+  # -- Additional envoy volumeMounts.
+  extraVolumeMounts: []
+
+  # -- Configure termination grace period for cilium-envoy DaemonSet.
+  terminationGracePeriodSeconds: 1
+
+  # -- TCP port for the health API.
+  healthPort: 9878
+
+  updateStrategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxUnavailable: 2
+  # -- Roll out cilium envoy pods automatically when configmap is updated.
+  rollOutPods: false
+
+  # -- Security Context for cilium-envoy pods.
+  podSecurityContext: {}
+
+  # -- Annotations to be added to envoy pods
+  podAnnotations: {}
+
+  # -- Labels to be added to envoy pods
+  podLabels: {}
+
+  # -- Envoy resource limits & requests
+  # ref: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/
+  resources: {}
+    # limits:
+    #   cpu: 4000m
+    #   memory: 4Gi
+    # requests:
+    #   cpu: 100m
+    #   memory: 512Mi
+
+  startupProbe:
+    # -- failure threshold of startup probe.
+    # 105 x 2s translates to the old behaviour of the readiness probe (120s delay + 30 x 3s)
+    failureThreshold: 105
+    # -- interval between checks of the startup probe
+    periodSeconds: 2
+  livenessProbe:
+    # -- failure threshold of liveness probe
+    failureThreshold: 10
+    # -- interval between checks of the liveness probe
+    periodSeconds: 30
+  readinessProbe:
+    # -- failure threshold of readiness probe
+    failureThreshold: 3
+    # -- interval between checks of the readiness probe
+    periodSeconds: 30
+
+  securityContext:
+    # -- User to run the pod with
+    # runAsUser: 0
+    # -- Run the pod with elevated privileges
+    privileged: false
+    # -- SELinux options for the `cilium-envoy` container
+    seLinuxOptions:
+      level: 's0'
+      # Running with spc_t since we have removed the privileged mode.
+      # Users can change it to a different type as long as they have the
+      # type available on the system.
+      type: 'spc_t'
+    capabilities:
+      # -- Capabilities for the `cilium-envoy` container
+      envoy:
+        # Used since cilium proxy uses setting IPPROTO_IP/IP_TRANSPARENT
+        - NET_ADMIN
+        # We need it for now but might not need it for >= 5.11 specially
+        # for the 'SYS_RESOURCE'.
+        # In >= 5.8 there's already BPF and PERMON capabilities
+        - SYS_ADMIN
+        # Both PERFMON and BPF requires kernel 5.8, container runtime
+        # cri-o >= v1.22.0 or containerd >= v1.5.0.
+        # If available, SYS_ADMIN can be removed.
+        #- PERFMON
+        #- BPF
+
+  # -- Affinity for cilium-envoy.
+  affinity:
+    podAntiAffinity:
```

Comment:
This is a daemonset; there's no need for anti-affinity.

Reply:
@squeed `affinity` is supported by the `DaemonSet` too - and again already supported for the Cilium Agent DaemonSet.

Reply:
We really don't want people specifying affinity here if possible, since we might want to use it to exclude nodes with embedded envoy enabled :-). I realize that rollout is later, but let's leave the avenue open.

Reply:
The default should still be empty; the antiAffinity here is just one-per-node, which the DaemonSet controller already does by default.

Plus, later, we can use the affinity here to control rollout from embedded to non-embedded. Just need a few cute label tricks :-)

Reply:
@squeed the antiAffinity is there because the DaemonSet doesn't guarantee that there will be only one pod per node. It was added by this PR https://github.com/cilium/cilium/pull/11830. IIRC I think this happened during an upgrade under specific certain conditions.

Reply:
wow, that's some subtle madness.